src/formatDate.js

```diff
@@ -9,6 +9,34 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const result = Array(3).fill(0);
+  const dateArray = date.split(fromFormat.slice(-1));
```

Comment:
The delimiter used in `date.split(fromFormat.slice(-1))` might not correctly split the date string if the delimiter is not the last character of `fromFormat`. Consider explicitly defining the delimiter or extracting it from the format string.